### PR TITLE
Reformat code with JuliaFormatter

### PR DESCRIPTION
**Pull request by devmotion**
JuliaFormatter will ensure that the Julia code is formatted consistently. I also added an action that checks the format and makes suggestions for how to fix the formatting.

### CODE REVIEW

File: .format/Project.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[deps]
+JuliaFormatter = "98e50ef6-434e-11e9-1051-2b60c6c9e899"
+
+[compat]
+JuliaFormatter = "1"
```
Comment (DilumAluthge) [inline .format/Project.toml]: Can you make sure that this file ends in a newline character?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This is the default target
+.PHONY: default
+default:
+	$(error You must provide a target)
+
+### JuliaFormatter
+.PHONY: format
+format:
```
Comment (DilumAluthge) [inline Makefile]: Just to make it obvious that this command will modify the files in place, can you rename it something more explicit, like `format-fix` or something like that?

**Review by DilumAluthge — CHANGES_REQUESTED** [review]
Just two small changes.

File: .format/Project.toml (hunk 1 above)
Comment (devmotion) [reply .format/Project.toml]: Done.

File: Makefile (hunk 2 above)
Comment (DilumAluthge) [reply Makefile]: This is done now.

Comment (devmotion) [reply Makefile]: I renamed it to `format-fix`.

**Review by DilumAluthge — APPROVED** [review]
Thank you David!

**Comment by DilumAluthge** [comment]
This is good to merge from my POV.

**Pull request merged**